### PR TITLE
feat(redis): upgrade to use Redis 7

### PR DESCRIPTION
**Pull request by rafaelromcar-parabol**
# Description

We are already using Redis 7 in both Digital Ocean and GCP. This PR upgrades all Redis references to 7.0.

- It uses 7.0, without a patch version, because it is what's configured in GCP. We cannot control the patch upgrades. By definition, these images aren't inmutable, but the risk is very low and we need to test things ahead of GCP automated upgrades.
- The image  [Docker official 7.0-alpine](https://hub.docker.com/_/redis/tags?page=1&name=7.0-alpine) is used everywhere.
- Redis TLS documentation has been modified in order to ease the use of that functionality in development if required.

## Testing scenarios

- [x] Build action works and produces a Docker image.
- [x] Developers can execute their local environment without problem.

### CODE REVIEW

**Comment by rafaelromcar-parabol** [comment]
Build action works and produces the Docker image. See [here](https://github.com/ParabolInc/parabol/actions/runs/6531285688/job/17732222082)

File: docker/dev.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@ services:
       - parabol-network
     restart: unless-stopped
   redis:
-    image: bitnami/redis:6.2
+    image: bitnami/redis:7.0-debian-11
```
Comment (Dschoordsch) [inline docker/dev.yml:54]: -1 I don't see why we should use the debian image locally when we can get away with the smaller alpine everywhere else. I don't quite understand the TLS argument either as we're not using TLS locally in any of our environments.

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
`yarn dev` locally works fine with the new image

File: docker/dev.yml (hunk 1 above)
Comment (rafaelromcar-parabol) [reply docker/dev.yml:54]: We could just rollback to the basic image using `7.0-alpine` but if ever we need to test or use the TLS feature, we would need to adapt the configuration for it. See [this](https://github.com/ParabolInc/parabol/pull/8488#issuecomment-1701209184) and [this](https://github.com/ParabolInc/parabol/pull/8488#issuecomment-1701546727).

That said, I think you are right and I'm changing it to redis:7.0-alpine. If we ever need to test that again, we will see how to do so.

File: docker/dev.yml [hunk 2]
```diff
@@ -51,7 +51,7 @@ services:
       - parabol-network
     restart: unless-stopped
   redis:
-    image: bitnami/redis:6.2
+    image: bitnami/redis:7.0-alpine
```
Comment (Dschoordsch) [inline docker/dev.yml:54]: -1 please remove `bitnami/`

**Review by Dschoordsch — CHANGES_REQUESTED** [review]

File: docker/dev.yml (hunk 2 above)
Comment (rafaelromcar-parabol) [reply docker/dev.yml]: Sorry, I was doing more changes. You can test now, and check the modified doc as well.

**Review by Dschoordsch — APPROVED** [review]

**Comment by rafaelromcar-parabol** [comment]
I assume you checked `yarn dev` back again @Dschoordsch ?

**Comment by Dschoordsch** [comment]
Yes, I did.

**Pull request merged**